src/header.jl

```diff
@@ -383,6 +386,37 @@ function setindex!(hdr::FITSHeader, value::Any, i::Integer)
     hdr.values[i] = value
 end
 
+
+"""
+    delete!(hdr::FITSHeader, key::String)
+
+Delete a key in a FITS header. 
+"""
+function delete!(hdr::FITSHeader, key::String)
+    if (haskey(hdr, key))
+        index = hdr.map[key]
+
+        # Delete the entries
+        deleteat!(hdr.values, index)
+        deleteat!(hdr.keys, index)
+        deleteat!(hdr.comments, index)
+
+        # Then re-create the map array
+        map = Dict{String, Int}()
+        for i in 1:length(hdr.keys)
+            map[hdr.keys[i]] = i
+        end
+        hdr.map = map
+
+    else
+        println("No key $key found!")
+    end
```

Comment:
I would recommend throwing an error instead of just printing a line, especially to stay in line with what you would expect from a Base Dictionary, since we're using that API.
```suggestion
    haskey(hdr, key) || throw(KeyError(key))
    
    index = hdr.map[key]

    # Delete the entries
    deleteat!(hdr.values, index)
    deleteat!(hdr.keys, index)
    deleteat!(hdr.comments, index)

    # Then re-create the map array
    map = Dict{String, Int}()
    for i in 1:length(hdr.keys)
        map[hdr.keys[i]] = i
    end
    hdr.map = map

```